financial-services-accelerator/components/org.wso2.financial.services.accelerator.gateway/src/main/java/org/wso2/financial/services/accelerator/gateway/executor/impl/consent/ConsentEnforcementExecutor.java

```diff
@@ -0,0 +1,318 @@
+                if (key == null) {
+                    try (FileInputStream is = new FileInputStream(dataHolder.getKeyStoreLocation())) {
+                        KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+                        keystore.load(is, dataHolder.getKeyStorePassword());
+                        key = keystore.getKey(dataHolder.getKeyAlias(), dataHolder.getKeyPassword().toCharArray());
+                    } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException
+                             | UnrecoverableKeyException e) {
+                        log.error("Error occurred while retrieving private key from keystore ", e);
+                    }
+                }
+            }
+        }
+        return key;
+    }
+
+    /**
+     * Method to generate JWT.
+     * @param payload Payload to be signed
+     * @return Signed JWT
+     */
+    protected String generateJWT(String payload) {
+
+        return Jwts.builder()
+                .setPayload(payload)
+                .signWith(SignatureAlgorithm.RS512, getJWTSigningKey())
+                .compact();
+    }
+
+    /**
+     * Method to invoke consent validation service when the JWT payload is provided.
+     *
+     * @param enforcementJWTPayload JWT Payload
+     * @return Response as a String
+     * @throws IOException When failed to invoke the validation endpoint or failed to parse the response.
+     */
+    @Generated(message = "Ignoring from unit tests since this method require calling external component to function")
+    private String invokeConsentValidationService(String enforcementJWTPayload) throws IOException,
+            FinancialServicesException {
+
+        HttpPost httpPost = new HttpPost(getValidationEndpoint());
+        StringEntity params;
+        params = new StringEntity(enforcementJWTPayload);
+        httpPost.setEntity(params);
+        httpPost.setHeader(GatewayConstants.CONTENT_TYPE_TAG, GatewayConstants.JWT_CONTENT_TYPE);
+        String userName = GatewayUtils.getAPIMgtConfig(GatewayConstants.API_KEY_VALIDATOR_USERNAME);
+        String password = GatewayUtils.getAPIMgtConfig(GatewayConstants.API_KEY_VALIDATOR_PASSWORD);
+        httpPost.setHeader(GatewayConstants.AUTH_HEADER, GatewayUtils.getBasicAuthHeader(userName, password));
+        CloseableHttpResponse response = HTTPClientUtils.getHttpsClient().execute(httpPost);
+        InputStream in = response.getEntity().getContent();
+        return IOUtils.toString(in, String.valueOf(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Method to handle errors.
+     *
+     * @param fsapiRequestContext API Context
+     * @param errorCode           Error Code
+     * @param errorMessage        Error Message
+     * @param httpCode            HTTP status code ( in 4XX range)
+     */
+    protected void handleError(FSAPIRequestContext fsapiRequestContext, String errorCode, String errorMessage,
```

Comment:
```suggestion
    protected void handleError(FSAPIRequestContext fsApiRequestContext, String errorCode, String errorMessage,
```

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/130/commits/edb99723b666242ae3b634377232e41d1a4105f9